### PR TITLE
Remove support for lenient booleans.

### LINKED ISSUE #22298: Cleanup migration code for strict booleans
In course of #22200, we add code to ease the migration to strict booleans.  In 7.0, we should remove the respective code again (all lenient parsing of booleans for BWC, deprecation logging, ...)

### PR DESCRIPTION
**Pull request by jpountz**
Closes #22298

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/analysis/SynonymGraphTokenFilterFactory.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public String name() {
         @Override
         public TokenStream create(TokenStream tokenStream) {
             // fst is null means no synonyms
-            return synonymMap.fst == null ? tokenStream : new SynonymGraphFilter(tokenStream, synonymMap, ignoreCase);
+            return synonymMap.fst == null ? tokenStream : new SynonymGraphFilter(tokenStream, synonymMap, false);
```
Comment (danielmitterdorfer) [inline core/src/main/java/org/elasticsearch/index/analysis/SynonymGraphTokenFilterFactory.java]: Hmm, I see that the field `ignoreCase` is deprecated but this change looks unrelated to this PR? Looks like you removed this field together with this PR (I think a separate commit would have been better but it's fine with me).

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for doing this cleanup @jpountz. 

LGTM, I left just one comment.

File: core/src/main/java/org/elasticsearch/index/analysis/SynonymGraphTokenFilterFactory.java (hunk 1 above)
Comment (jpountz) [reply core/src/main/java/org/elasticsearch/index/analysis/SynonymGraphTokenFilterFactory.java]: I'll split into a separate commit
